fe/fe-core/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java

```diff
@@ -342,6 +339,344 @@ public void updateDatabaseUsedQuotaData(long usedQuotaDataBytes) {
+                // partition maybe dropped between pre-commit and commit, ignore this error
+                if (partition == null) {
+                    partitionCommitInfoIterator.remove();
+                    LOG.warn("partition {} is dropped, skip version check and remove it from transaction state {}",
+                            partitionId,
+                            transactionState);
+                    continue;
+                }
+
+                int quorumReplicaNum = partitionInfo.getReplicaAllocation(partitionId).getTotalReplicaNum() / 2 + 1;
+
+                List<MaterializedIndex> allIndices;
+                if (transactionState.getLoadedTblIndexes().isEmpty()) {
+                    allIndices = partition.getMaterializedIndices(MaterializedIndex.IndexExtState.ALL);
+                } else {
+                    allIndices = Lists.newArrayList();
+                    for (long indexId : transactionState.getLoadedTblIndexes().get(tableId)) {
+                        MaterializedIndex index = partition.getIndex(indexId);
+                        if (index != null) {
+                            allIndices.add(index);
+                        }
+                    }
+                }
+
+                for (MaterializedIndex index : allIndices) {
+                    for (Tablet tablet : index.getTablets()) {
+                        int healthReplicaNum = 0;
+                        for (Replica replica : tablet.getReplicas()) {
+                            List<Long> committedTxnIds = replica.getCommittedTxnIds();
+                            if (committedTxnIds.contains(transactionId)) {
+                                ++healthReplicaNum;
+                            } else {
+                                errorReplicaIds.add(replica.getId());
+                            }
+                        }
+
+                        if (healthReplicaNum < quorumReplicaNum) {
+                            LOG.warn("failed to commit transaction {} on tablet {}, with only {} replicas less than quorum {}, txnId: {}",
+                                    transactionState, tablet, healthReplicaNum, quorumReplicaNum, transactionId);
+                            throw new TransactionCommitFailedException("tablet [" + tablet.getId() +"], with only ["
+                                    +healthReplicaNum+ "] replicas less than quorum [" +quorumReplicaNum+ "], txnId ["
+                                    + transactionId + "]");
+                        }
+                    }
+                }
+            }
+        }
+        transactionState.setErrorReplicas(errorReplicaIds);
+
+        return transactionState;
+    }
+
+    public void commitTransaction(List<Long> transactionIds) throws UserException {
+        try {
+            List<TransactionState> transactionStates = Lists.newArrayList();
+            for (long transactionId : transactionIds) {
+                transactionStates.add(checkPreCommitStatus(transactionId));
```

Comment:
If the txn is in PRECOMMITTED, it should be able to be COMMITTED, why we still need to call `checkPreCommitStatus()` here?
The only thing need to be done in `checkPreCommitStatus()` is to check whether the txn state is PRECOMMITTED, but I saw that there other checking list in `checkPreCommitStatus()`, why?